internal/service/adapters.go

```diff
@@ -0,0 +1,147 @@
+	kubernetesruntime "github.com/topfreegames/maestro/internal/adapters/runtime/kubernetes"
+	portallocatorrandom "github.com/topfreegames/maestro/internal/adapters/port_allocator/random"
+	schedulerstoragepg "github.com/topfreegames/maestro/internal/adapters/scheduler_storage/pg"
+	"github.com/topfreegames/maestro/internal/config"
+	"github.com/topfreegames/maestro/internal/core/ports"
+	"github.com/topfreegames/maestro/internal/core/entities"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// configurations paths for the adapters
+const (
+	// Kubernetes runtime
+	runtimeKubernetesMasterUrlPath  = "adapters.runtime.kubernetes.masterUrl"
+	runtimeKubernetesKubeconfigPath = "adapters.runtime.kubernetes.kubeconfig"
+	// Redis operation storage
+	operationStorageRedisUrlPath = "adapters.operationStorage.redis.url"
+	// Redis room storage
+	roomStorageRedisUrlPath = "adapters.roomStorage.redis.url"
+	// Redis instance storage
+	instanceStorageRedisUrlPath = "adapters.intanceStorage.redis.url"
+	instanceStorageRedisScanSizePath = "adapters.intanceStorage.redis.scanSize"
+	// Random port allocator
+	portAllocatorRandomRangePath = "adapters.portallocator.random.range"
+	// Postgres scheduler storage
+	schedulerStoragePostgresUrlPath = "adapters.schedulerStorage.postgres.url"
+)
+
+func NewRuntimeKubernetes(ctx context.Context, c config.Config) (ports.Runtime, error) {
+	clientset, err := createKubernetesClient(
+		ctx,
+		c.GetString(runtimeKubernetesMasterUrlPath),
+		c.GetString(runtimeKubernetesKubeconfigPath),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize Kubernetes runtime: %w", err)
+	}
+
+	return kubernetesruntime.New(clientset), nil
+}
+
+func NewOperationStorageRedis(ctx context.Context, clock ports.Clock, c config.Config) (ports.OperationStorage, error) {
+	client, err := createRedisClient(ctx, c.GetString(operationStorageRedisUrlPath))
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize Redis operation storage: %w", err)
+	}
+
+	return operationstorageredis.NewRedisOperationStorage(client, clock), nil
+}
+
+func NewRoomStorageRedis(ctx context.Context, c config.Config) (ports.RoomStorage, error) {
+	client, err := createLegacyRedisClient(ctx, c.GetString(roomStorageRedisUrlPath))
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize Redis room storage: %w", err)
+	}
+
+	return roomstorageredis.NewRedisStateStorage(client), nil
+}
+
+func NewGameRoomInstanceStorageRedis(ctx context.Context, c config.Config) (ports.GameRoomInstanceStorage, error) {
+	client, err := createLegacyRedisClient(ctx, c.GetString(instanceStorageRedisUrlPath))
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize Redis instance storage: %w", err)
+	}
+
+	return instancestorageredis.NewRedisInstanceStorage(client, c.GetInt(instanceStorageRedisScanSizePath)), nil
+}
+
+func NewClockTime() (ports.Clock, error) {
+	return clocktime.NewClock(), nil
```

Comment:
Yeah, it makes sense to me, I was trying to keep the same shape on the returns, but since we're not doing it on the arguments, I think it's ok to change that. I'll also drop the unnecessary `context.Context`.